src/requirements.txt

```diff
@@ -10,6 +10,7 @@ Jinja2==3.0.3
 Flask==1.1.2
 flask_cors==3.0.10
 globus_sdk==2.0.1
+hubmap-sdk==1.0.1
```

Comment:
I'm not sure about using the hubmap-sdk, just to retrieve the dataset entity. @ChuckKollar had created the feign api client earlier to facilitate: https://github.com/hubmapconsortium/ingest-api/blob/test-release/src/api/entity_api.py#L18 

@shirey any suggestions?

Reply:
Regarding using hubmap-sdk, I can certainly eschew it in favor of chuck's api client. I was under the impression that, broadly speaking, the goal was to eventually be able to use the hubmap-sdk throughout all of our webservices, particularly if it eventually grows to include more than just search and entity. I'll leave that decision to you guys. 

As far as putting a try block around doi draft creation, I certainly think thats a good idea, but it actually brings me to another issue. 

If a doi draft fails for whatever reason, the dataset.status will already be changed to published, which can't be directly changed back through entity api, and the script that handles the doi stuff performs a check to verify that its a published dataset. I wonder, perhaps, if it might be appropriate to wait to set dataset.status to published until after doi successfully creates the draft? Or maybe there's reasons this cannot be the case.

Reply:
That's good thoughts. We can publish a dataset first by changing the status to "Published" and this does not rely on the DOI. To me they are separate, but making the dataset publish is always the first step. 

The reason of adding try/except is if a DOI fails to create due to the DataCite API failure or network issue, the publish api call stops, and that's the time we'll need to manually resolve the data issue directly with DataCite or try again.